environment_test.go

```diff
@@ -489,3 +489,53 @@ func TestEnvironment_Time(t *testing.T) {
 			})
 	}
 }
+
+func TestEnvironment_List(t *testing.T) {
+	env := newEnvironment(newMockChain(t))
+
+	assert.Equal(t, []string{}, env.List())
+
+	env.Put("k1", 1)
+	env.Put("k2", 2)
+	env.Put("k3", 3)
+	assert.Equal(t, []string{"k1", "k2", "k3"}, env.List())
+
+	env.Put("abc", 4)
+	assert.Equal(t, []string{"abc", "k1", "k2", "k3"}, env.List())
+
+	env.Delete("k2")
+	env.Delete("k3")
+	env.Put("pqr", 5)
+	assert.Equal(t, []string{"abc", "k1", "pqr"}, env.List())
+}
+
+func TestEnvironment_Match(t *testing.T) {
```

Comment:
Could you please divide this test into two: one for invalid glob pattern, another for normal checks? (We use same approach in most tests).

Reply:
Split the test into two subtests.